hooks/useImageData.tsx

```diff
@@ -24,7 +24,7 @@ const useImageData = (tokenIds: string[]) => {
     } else {
       setImageUris(Array(tokenIds.length).fill(FALLBACK_IMAGE));
     }
-  }, [data, tokenIds]);
```

Comment:
this seems to be the source of the re-rendering. so the tokenIds dependency can be removed or better still memoized